services/frontend/adapters/inbound/k8s_sidecar_notifications_receiver.adapter.go

```diff
@@ -0,0 +1,106 @@
+package inboundAdapters
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/berops/claudie/services/frontend/domain/usecases"
+)
+
+// Receives notifications from the k8s-sidecar container running in the same pod
+// The notifications are regarding new/updated claudie manifests
+type K8sSidecarNotificationsReceiver struct {
+	usecases *usecases.Usecases
+
+	// manifestDir represents the path to the directory containing claudie manifest files which will be
+	// watched by the k8s-sidecar container.
+	manifestDir string
+
+	// server is the underlying HTTP server which receives notifications
+	// from the k8s sidecar container when the manifest files are created / updated /deleted.
+	server *http.Server
+
+	// waitGroup is used to handle a graceful shutdown of the HTTP server.
+	// It will wait for all spawned go-routines to finish their work.
+	waitGroup sync.WaitGroup
+}
+
+func NewK8sSidecarNotificationsReceiver(usecases *usecases.Usecases) (*K8sSidecarNotificationsReceiver, error) {
+
+	k8sSidecarNotificationsReceiver := &K8sSidecarNotificationsReceiver{
+
+		manifestDir: os.Getenv("MANIFEST_DIR"),
+		usecases:    usecases,
+
+		server: &http.Server{ReadHeaderTimeout: 2 * time.Second},
+	}
+
+	k8sSidecarNotificationsReceiver.registerNotificationHandlers()
+
+	return k8sSidecarNotificationsReceiver, k8sSidecarNotificationsReceiver.PerformHealthCheck()
+}
+
+func (k *K8sSidecarNotificationsReceiver) registerNotificationHandlers() {
+
+	var router *http.ServeMux = http.NewServeMux()
+
+	// TODO: rename the route "/reload" to "/process-manifest-files"
+	router.HandleFunc("/reload", k.processManifestFilesHandler)
```

Comment:
No, this PR is regarding the implementation of clean architecture :). So let's stick to that. I wrote that comment so that we can discuss about it (and implement it if we agree) after this PR gets merged.